src/core/coap/secure_coap_server.cpp

```diff
@@ -117,7 +117,8 @@ void SecureServer::Receive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
     {
         mPeerAddress.SetPeerAddr(aMessageInfo.GetPeerAddr());
         mPeerAddress.SetPeerPort(aMessageInfo.GetPeerPort());
-
+        mPeerAddress.SetSockAddr(aMessageInfo.GetSockAddr());
```

Comment:
In Thread Network, both the Commissioner and Border Agent need to start secure CoAP server, so the Commissioner may also face the same problem, not just the Border Agent.